servicetalk-concurrent-internal/src/main/java/io/servicetalk/concurrent/internal/ConcurrentUtils.java

```diff
@@ -78,6 +79,36 @@ private ConcurrentUtils() {
         return lockUpdater.getAndSet(owner, CONCURRENT_IDLE) == CONCURRENT_EMITTING;
     }
 
+    /**
+     * Acquire a lock that is exclusively held with no re-entry, but attempts to acquire the lock while it is
+     * held can be detected by {@link #releaseLock(AtomicInteger)}.
+     * @param lock The {@link AtomicInteger} used to control the lock state.
+     * @return {@code true} if the lock was acquired, {@code false} otherwise.
+     */
+    public static boolean tryAcquireLock(AtomicInteger lock) {
```

Comment:
If we use named `Runnable`(s) (static inner classes) in `PublisherFlatMapConcatUtils` we can reuse existing utilities that operate with `AtomicIntegerFieldUpdater` and save some memory.

Reply:
good catch! named `Function` will do the trick in this case based upon where state lives.